portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +30,54 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private List<String> messages = new ArrayList<>();
+  private String json;
+
+  /** Converts Messages ArrayList to Json */
+  public void toJson() throws IOException {
+    json = new Gson().toJson(messages);
+  }
+
+  /** Redirection to the main page */
+  private void doRedirect(HttpServletResponse response) throws IOException {
+    response.sendRedirect(
+      "/../../../../../../index.html"
+    );
+  }
+
+  /** Write a new message to DataStore */
+  private void writeMessageToDatastore(String message) throws IOException {
+    Entity messageEntity = new Entity("Message");
+
+    messageEntity.setProperty("message", message);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(messageEntity);
+  }
+
+  private void dataServletResponse(HttpServletResponse response) throws IOException {
+    response.setContentType("text/html;");
+    response.getWriter().println(json);
+  }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String value = request.getParameter("user-comment");
+   
+    /** Redirection if value is empty or accidental click */
+    if(value == null || value.length() == 0) {
+      doRedirect(response);
```

Comment:
Should you return after doRedirect()? Otherwise, this function will continue to execute and call dataServletResponse() and doRedirect() again.

Reply:
Sounds like a good idea, even though those actions are made to refresh the webpage, it certainly can save some processes with the return